v-next/hardhat-ignition/src/internal/hook-handlers/config.ts

```diff
@@ -0,0 +1,70 @@
+import type {
+  ConfigurationVariableResolver,
+  HardhatConfig,
+  HardhatUserConfig,
+  IgnitionUserConfig,
+} from "@ignored/hardhat-vnext/types/config";
+import type { ConfigHooks } from "@ignored/hardhat-vnext/types/hooks";
+
+import { assertHardhatInvariant } from "@ignored/hardhat-vnext-errors";
+import { resolveFromRoot } from "@ignored/hardhat-vnext-utils/path";
+
+export default async (): Promise<Partial<ConfigHooks>> => ({
+  resolveUserConfig,
+});
+
+export async function resolveUserConfig(
+  userConfig: HardhatUserConfig,
+  resolveConfigurationVariable: ConfigurationVariableResolver,
+  next: (
+    nextUserConfig: HardhatUserConfig,
+    nextResolveConfigurationVariable: ConfigurationVariableResolver,
+  ) => Promise<HardhatConfig>,
+): Promise<HardhatConfig> {
+  const resolvedConfig = await next(userConfig, resolveConfigurationVariable);
+
+  const updatedNetworks = Object.fromEntries(
+    Object.entries(resolvedConfig.networks).map(
+      ([networkName, networkConfig]) => {
+        assertHardhatInvariant(
+          userConfig.networks !== undefined,
+          "Networks are undefined",
+        );
+
+        const givenIgnition = userConfig.networks[networkName].ignition ?? {};
+
+        return [
+          networkName,
+          {
+            ...networkConfig,
+            ignition: {
+              maxFeePerGasLimit: givenIgnition.maxFeePerGasLimit,
+              maxPriorityFeePerGas: givenIgnition.maxPriorityFeePerGas,
+              gasPrice: givenIgnition.gasPrice,
+              disableFeeBumping: givenIgnition.disableFeeBumping,
+              explorerUrl: givenIgnition.explorerUrl,
+            },
+          },
+        ];
+      },
+    ),
+  );
+
+  const paths = resolvedConfig.paths ?? {};
+  const ignition: IgnitionUserConfig = userConfig.ignition ?? {};
+
+  return {
+    ...resolvedConfig,
+    networks: updatedNetworks,
+    paths: {
+      ...paths,
+      ignition: resolveFromRoot(
+        resolvedConfig.paths.root,
+        paths.ignition ?? "ignition",
```

Comment:
ignition will never be here, as it wasn't resolved by core. It should be fetched from the user config.

Reply:
My mental model of this change:
1. `resolveUserConfig` is a hook that injects into the chain that resolves user config into Hardhat config
2. the `resolvedConfig` here is coming from early links in the chain and is hence the Hardhat config in the process of being built up - not a direct copy of the user config.
3. The intent here is a HardhatUser config with a populated `paths.ignition` entry. To do that I should use the passed in `userConfig.paths.ignition` and fallback to the `"ignition"` default otherwise.

Reply:
I have updated the line at 62 to:
 
```ts
userConfig.paths?.ignition ?? "ignition",
```